feat(scroll-area): add raw prop to disable Radix wrapper via asChild

Harshul23 opened the pull request:
## Summary

This PR introduces an optional raw prop to the ScrollArea component, allowing developers to render the viewport’s children directly without the internal Radix wrapper    

`<div style="min-width: 100%; display: table;"> `

This solves layout issues where the default wrapper conflicts with flex or grid-based children (e.g., in responsive or dynamic UIs).

## Motivation

By default, Radix UI’s ScrollArea.Viewport adds a structural wrapper that enforces a table display and min-width of 100%.
This behavior is fine for most cases, but it restricts flexibility for advanced layouts like flex rows or grids inside scrollable containers.

Developers have raised this issue in:
- [#8342](https://github.com/shadcn-ui/ui/issues/8342)
- [#3833](https://github.com/shadcn-ui/ui/issues/3833)

Adding this prop provides control while maintaining backward compatibility.

## Changes
- Added raw?: boolean prop to ScrollArea.
- Passed raw to Radix’s Viewport via asChild={raw}.
- Added prop documentation comment.
- Default remains unchanged (raw = false).

### Example Usage

Before (default behavior)
```
<ScrollArea className="h-40 w-40 border">
  <div className="flex gap-4">
    <div className="w-20 h-20 bg-red-500" />
    <div className="w-20 h-20 bg-blue-500" />
  </div>
</ScrollArea>
```
❌ Wrapped by Radix → display: table; min-width: 100% breaks flex layout.


After (with raw)
```
<ScrollArea raw className="h-40 w-40 border">
  <div className="flex gap-4">
    <div className="w-20 h-20 bg-red-500" />
    <div className="w-20 h-20 bg-blue-500" />
  </div>
</ScrollArea>
```
✅ Renders children directly, preserving flex/grid behavior.

Checklist
- [x] Added prop type and inline docs
- [x] 	Backward compatible
- [x] 	Tested with flex, grid, and content-overflow cases
- [x] 	Linked related issues (#8342)

### Notes for Maintainers

> This implementation leverages Radix’s asChild prop, aligning with Radix best practices.
> No breaking changes or additional dependencies introduced.

sawa-ko commented:
The scroll stops working.

```jsx
<ScrollArea raw={true} className={"h-[calc(100vh-5rem)] w-[75%]"}>
          <div className="flex flex-col gap-4">
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
            <div className="w-20 h-20 bg-red-500" />
            <div className="w-20 h-20 bg-blue-500" />
          </div>
</ScrollArea>
```

Harshul23 commented:
Hi! @sawa-ko  I’ve fixed and resolved all merge conflicts in this PR and implemented the raw behavior via asChild on the Viewport. Sorry for the earlier bad merge/commit.

Summary:

Removed conflict markers in apps/v4/registry/new-york-v4/ui/scroll-area.tsx
Kept the asChild={raw} approach (no cloneElement), with a shared viewportClassName
Backward compatible, raw defaults to false.
Commit: a3565f6046967b001a74491b4d724a4598188455

Please review, happy to adjust or revert changes if needed.
